lldb/unittests/Symbol/TestSwiftASTContext.cpp

```diff
@@ -199,3 +199,32 @@ TEST_F(TestSwiftASTContext, IsNonTriviallyManagedReferenceType) {
                                                                    nullptr));
 #endif
 }
+
+namespace {
+  const std::vector<std::string> duplicated_flags = {
+    "-DMACRO1", "-D", "MACRO1", "-UMACRO2", "-U", "MACRO2",
+    "-I/path1", "-I", "/path1", "-F/path2", "-F", "/path2",
+    "-fmodule-map-file=/path3", "-fmodule-map-file=/path3",
+    "-F/path2", "-F", "/path2", "-I/path1", "-I", "/path1",
+    "-UMACRO2", "-U", "MACRO2", "-DMACRO1", "-D", "MACRO1",
+  };
+  const std::vector<std::string> unqued_flags = {
+    "-DMACRO1", "-UMACRO2", "-I/path1", "-F/path2", "-fmodule-map-file=/path3"
```

Comment:
@adrian-prantl I've noticed that current uniquing mechanism assumes that multi-arg flags are already joined in `ClangImporterOptions`, and paths are absolute. Is it correct assumption, or should I fix it by processing existing flags same way as newly added?

Reply:
Are you talking about this function? https://github.com/apple/llvm-project/blob/swift/master/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp#L1433

Reply:
Exactly! Thanks for finding it for me.

Reply:
So I see that it is used for args that are added through `AddExtraClangArgs` method, but is it true that it is the only way how `ClangImporterOptions::ExtraFlags` gets populated with clang args? If it's not, there could still be some args duplication (however, nearly not as big as before)